src/event/event.controller.ts

```diff
@@ -60,6 +64,48 @@ export class EventController {
     }
   }
 
+  @Put(':id')
+  @UsePipes(
+    new ValidationPipe({
+      transform: true,
+    }),
+  )
+  @UseGuards(AuthGuard)
+  @UseInterceptors(
+    FileFieldsInterceptor([{ name: 'poster_images' }], {
+      fileFilter: (req, file, callback) => {
+        if (!ALLOWED_MIME_TYPES.includes(file.mimetype)) {
+          return callback(
+            new BadRequestException(
+              `Invalid file type: ${file.mimetype}. Only JPEG, PNG, and GIF are allowed.`,
+            ),
+            false,
+          );
+        }
+
+        callback(null, true);
+      },
+    }),
+  )
+  async update(
+    @Param('id', ParseIntPipe) id: number,
+    @Body() updateEventDto: UpdateEventDto,
+    @UploadedFiles() files: { poster_images?: Express.Multer.File[] },
+  ) {
+    try {
+      const updatedEvent = await this.eventService.updateEvent(
+        id,
+        updateEventDto,
+        files.poster_images,
+        updateEventDto.delete_poster_ids,
+      );
+
+      return { message: 'Event updated successfully', data: updatedEvent };
+    } catch (error) {
+      throw error;
+    }
```

Comment:
no need try catch in controller level. This should be done inside service. 
return the method of the event service lang along with arguments.